### PR TITLE
Bug 1904507: Add metrics for Datastore types

### PR DESCRIPTION
**Pull request by jsafrane**
NFS datastore type may be deprecated upstream and we want to know how many clusters use them.
(see https://vsphere-csi-driver.sigs.k8s.io/compatiblity_matrix.html)

The metric is collected in `CheckStorageClasses()`, therefore the default data store must be checked there if a StorageClass does not refer to a specific one. Extra care is made to collect each DS only once.

The simulated vCenter used for unit tests has all datastores with type `OTHER`. I did not change that to make sure the files can be re-generated later without manual changes.

Example:
```
# HELP vsphere_datastore_total [ALPHA] Number of DataStores used by the cluster.
# TYPE vsphere_datastore_total gauge
vsphere_datastore_total{type="vsan"} 2
```

### CODE REVIEW

**Comment by jsafrane** [comment]
cc @gnufied

File: pkg/check/datastore.go [hunk 1]
```diff
@@ -27,8 +30,48 @@ const (
 	dataCenterType        = "Datacenter"
 	DatastoreInfoProperty = "info"
 	SummaryProperty       = "summary"
+
+	dataStoreType = "type"
+)
+
+var (
+	dataStoreTypesMetric = metrics.NewGaugeVec(
```
Comment (gnufied) [inline pkg/check/datastore.go:38]: Did we really wanted this to be a gauge?

File: pkg/check/datastore.go [hunk 2]
```diff
@@ -52,18 +96,29 @@ func CheckStorageClasses(ctx *CheckContext) error {
...
+				// check the default datastore and collect its type.
+				if err := checkDefaultDatastoreWithDSType(ctx, dsTypes); err != nil {
+					klog.V(2).Infof("CheckStorageClasses: %s: %s", sc.Name, err)
+				}
 			}
 		}
 	}
+
+	for dsType, count := range dsTypes.getDataStoreTypeCount() {
+		dataStoreTypesMetric.WithLabelValues(dsType).Set(float64(count))
```
Comment (gnufied) [inline pkg/check/datastore.go:119]: nit: I think we should lowercase everything we get from vsphere APIs, so as not to introduce additional labels if something changes.

**Review by gnufied — COMMENTED** [review]
Mostly looks good. Some minor comment.

File: pkg/check/datastore.go (hunk 1 above)
Comment (jsafrane) [reply pkg/check/datastore.go:38]: All other metrics are gauges.

File: pkg/check/datastore.go (hunk 2 above)
Comment (jsafrane) [reply pkg/check/datastore.go:121]: Added lowercase coversion in `addDataStore()`, https://github.com/openshift/vsphere-problem-detector/pull/43/files#diff-4d023f9a9c662cddd2bcde080da1d94e60119d4f41fae7b286d368ebe9d3320fR60

**Comment by jsafrane** [comment]
/override ci/prow/e2e-vsphere
Known DNS bug: https://bugzilla.redhat.com/show_bug.cgi?id=1963912

**Pull request merged**